wlauto/workloads/adobereader/uiauto/app/src/main/java/com/arm/wlauto/uiauto/adobereader/UiAutomation.java

```diff
@@ -102,6 +102,10 @@ private void dismissWelcomeView() throws Exception {
         if (!onboarding_finish_button.exists()) {
             welcomeView.swipeLeft(10);
         }
+        
+        if (!onboarding_finish_button.exists()) {
+        return;
+        }
```

Comment:
I would suggest changing this to make the clicking of the UiButton conditional on its existence instead, as it would still allow the following checks to execute rather than returning early.
E.g something like:

```
if (onboarding_finish_button.exists()) {
    onboarding_finish_button.click()
}
```